src/vs/editor/common/modes.ts

```diff
@@ -392,7 +392,7 @@ export interface ReferenceProvider {
 	/**
 	 * Provide a set of project-wide references for the given position and document.
 	 */
-	provideReferences(model: editorCommon.IReadOnlyModel, position: Position, context: ReferenceContext, token: CancellationToken): Location[] | Thenable<Location[]>;
+	provideReferences(model: editorCommon.IReadOnlyModel, position: Position, context: ReferenceContext, progress: (value: Location) => void, token: CancellationToken): Location[] | Thenable<Location[]>;
```

Comment:
> This PR doesn't really wire up the streaming to any LS that supports streaming, right?

Correct.

> I think we should do that before proposing this to MS.

I don't agree. We can open this pull request with Microsoft (once we are happy) and start getting feedback early. Plus this PR will help them understand the streaming proposal itself. I agree that we/they might not want to merge this PR until LSP changes are finalized (because it may affect the implementation), but that doesn't mean we can't open it.

When dealing with external teams it is usually better to engage them sooner rather than later. Can you open an issue/PR with Microsoft about the LSP proposal?